microsoft/promptflow
# [Feature] Support default input index when collect package tools for custom-llm tool

# Description
### Reason
   Previously, customers complained that the order of custom-llm tool inputs displayed in the UI was inconsistent with the order defined in tool interface or yaml file. This was because the UI received the inputs as a dict, which resulted in an alphabetical order display.   
   In this PR, we will automatically add "ui_hints": {"index": XX} to every input based on the order in tool interface or yaml file for custom_llm tool. When UI received tool inputs, they can display the inputs according to this ui_hints["index"] field to make the order consistent. Both portal and extension have completed the code change to support ui_hints["index"].

### Cases which will not be corrected by this pr:
- Existing flow with flow.tools.json
- All examples in promptflow gallery which also have flow.tools.json file

### Test 
For more test cases, please go to [this link](https://microsoft.sharepoint.com/:o:/r/teams/STCASharedDataTeam/_layouts/15/Doc.aspx?sourcedoc=%7Bde8c0032-8386-492d-af0e-104af9de317a%7D&action=edit&wd=target(Pipelines%2FDCM%2FDevs%2Fjiazeng.one%7C0FA273ED-FDCC-4644-82B0-C96606E278CB%2FTest%20default%20input%20index-3%7CDC170A9F-4572-4931-BFAD-CEECD018616B%2F)&share=IgEyAIzehoMtSa8OEEr53jF6AYDaZzIchXLytsUQwUbc29U)
- custom-llm tool
![image](https://github.com/microsoft/promptflow/assets/95729303/942b03a2-ae12-44bb-ba03-ccad98df5c69)
    - in portal
![image](https://github.com/microsoft/promptflow/assets/95729303/2f5cd399-de70-4eb9-b26f-cc44fd9f45c5)
    - in extension
![image](https://github.com/microsoft/promptflow/assets/95729303/f22e2b6f-71df-4787-bab0-16645f9eb264)

- built in custom-llm tool
    - in portal:
![image](https://github.com/microsoft/promptflow/assets/95729303/98faab67-e59a-4e48-b843-58f373638cd5)
    - in extension:
![image](https://github.com/microsoft/promptflow/assets/95729303/f5311e36-beb8-4be7-ba29-abec5148f2c7)

- Test cases
The behavior of the following test cases are as expected. Only custom_llm tool's inputs' order are corrected, other tool's inputs' order keep same as before.
    - Create a new flow
    - Clone a flow
    - Upload a flow without .promptflow folder



  



# All Promptflow Contribution checklist:
- [ ] **Create an issue and link to the pull request to get dedicated review from promptflow team. Learn more: [suggested workflow](../CONTRIBUTING.md#suggested-workflow).**

## General Guidelines and Best Practices

### Testing Guidelines
